django/db/backends/sqlite3/base.py

```diff
@@ -451,39 +455,54 @@ def _sqlite_format_dtdelta(conn, lhs, rhs):
 
 
 def _sqlite_time_diff(lhs, rhs):
-    left = backend_utils.typecast_time(lhs)
-    right = backend_utils.typecast_time(rhs)
-    return (
-        (left.hour * 60 * 60 * 1000000) +
-        (left.minute * 60 * 1000000) +
-        (left.second * 1000000) +
-        (left.microsecond) -
-        (right.hour * 60 * 60 * 1000000) -
-        (right.minute * 60 * 1000000) -
-        (right.second * 1000000) -
-        (right.microsecond)
-    )
+    try:
+        left = backend_utils.typecast_time(lhs)
+        right = backend_utils.typecast_time(rhs)
+        return (
+            (left.hour * 60 * 60 * 1000000) +
+            (left.minute * 60 * 1000000) +
+            (left.second * 1000000) +
+            (left.microsecond) -
+            (right.hour * 60 * 60 * 1000000) -
+            (right.minute * 60 * 1000000) -
+            (right.second * 1000000) -
+            (right.microsecond)
+        )
+    except (AttributeError, TypeError, ValueError):
+        return None
 
 
 def _sqlite_timestamp_diff(lhs, rhs):
-    left = backend_utils.typecast_timestamp(lhs)
-    right = backend_utils.typecast_timestamp(rhs)
-    return duration_microseconds(left - right)
+    try:
+        left = backend_utils.typecast_timestamp(lhs)
+        right = backend_utils.typecast_timestamp(rhs)
+        return duration_microseconds(left - right)
+    except (TypeError, ValueError):
+        return None
 
 
 def _sqlite_regexp(re_pattern, re_string):
     return bool(re.search(re_pattern, str(re_string))) if re_string is not None else False
 
 
 def _sqlite_lpad(text, length, fill_text):
-    if len(text) >= length:
-        return text[:length]
-    return (fill_text * length)[:length - len(text)] + text
+    try:
+        if len(text) >= length:
+            return text[:length]
+        return (fill_text * length)[:length - len(text)] + text
+    except (TypeError, ValueError):
+        return None
 
 
 def _sqlite_rpad(text, length, fill_text):
-    return (text + fill_text * length)[:length]
+    try:
+        return (text + fill_text * length)[:length]
+    except (TypeError, ValueError):
+        return None
 
 
 def _sqlite_power(x, y):
-    return x ** y
+    try:
+        return x ** y
+    except (TypeError, ValueError):
+        return None
```

Comment:
This isn't correct as it will return `None` if `x` or `y` were a string.
It should only return `None` if x, y or both are `None`.